### PR TITLE
Feature/logger cleanup

### PR DESCRIPTION
**Pull request by pferrel**
Cleaning log messages

- in all cases it is know we need the engine-id in the message. Some errors may happen asynchronously so without the engine-id is is impossible to tell the affect of the error.
- we overuse `logger.info` and should use `logger.error` or `logger.warn` more often so log levels can be reduced without missing important messages

### CODE REVIEW

**Comment by pferrel** [comment]
not ready to merge, this is to keep track of ongoing logger message cleanup

File: rest-server/engines/src/main/scala/com/actionml/engines/ur/URPreparator.scala [hunk 1]
```diff
@@ -36,56 +36,27 @@ object URPreparator extends LazyLogging {
...
-      targetEntityType = Some(Some("item")))(sc).repartition(sc.defaultParallelism)
-    */
-
     // now separate the events by event name
     val indicatorNames = indicatorParams.map(_.name)
 
     logger.info(s"Indicators: ${indicatorNames}")
 
-    //val collectedEventsRDD = eventsRDD.collect()
+    //val debug = eventsRDD.collect()
```
Comment (andreyshikov) [inline rest-server/engines/src/main/scala/com/actionml/engines/ur/URPreparator.scala:44]: ^

**Review by qqmbr4k — APPROVED** [review]

**Comment by pferrel** [comment]
@andreyshikov this is a bunch of logger message cleanup. Conventions for messages:

- format of messages should include engine-id if present
- made as many info into trace as possible. Info should be things that are useful in a running system, trace are things that are usually not needed and may only show information about what harness or an engine is doing
- converted `debug` into `trace` for the most part

ready to merge

@andreyshikov you don't have to review all of these unless you want to.

File: rest-server/engines/src/main/scala/com/actionml/engines/ur/URPreparator.scala (hunk 1 above)
Comment (pferrel) [reply rest-server/engines/src/main/scala/com/actionml/engines/ur/URPreparator.scala:44]: @andreyshikov when you `.collect` any Spark data structure like an RDD it brings it all into memory in one process. This is ok for some debuging but very seldom used in real production Spark code. I have a convention of naming any collected thing "debug" so I can tell to not include them in production code. I would uncomment when using IntelliJ with small data. I have accidentally left these in when pushing so now I can just search for "debug" before pushing to make sure they are all taken out.

File: rest-server/conf/logback.xml [hunk 2]
```diff
@@ -22,21 +23,20 @@
...
-    </rollingPolicy>
-    <encoder>
-        <pattern>%d{HH:mm:ss.SSS} %-5level %-17logger{0} - %msg%n</pattern>
-    </encoder>
-</appender>
--->
+    <appender name="FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <file>${HARNESS_LOG_PATH}/harness.log</file>
+        <rollingPolicy class="ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy">
+            <fileNamePattern>harness-%d{yyyy-MM-dd}.%i.log</fileNamePattern>
```
Comment (andreyshikov) [inline rest-server/conf/logback.xml:31]: yes, that's the place that I was talking about. This path is not the same as `${HARNESS_LOG_PATH}/harness.log`

**Review by qqmbr4k — CHANGES_REQUESTED** [review]

File: rest-server/conf/logback.xml (hunk 2 above)
Comment (pferrel) [reply rest-server/conf/logback.xml:31]: @andreyshikov this is the old config, yours is commented out.

If there is an error in this line what should it be?

Comment (andreyshikov) [reply rest-server/conf/logback.xml:31]: `${HARNESS_LOG_PATH}/harness-%d{yyyy-MM-dd}.%i.log` if you want them in the same dir

Comment (pferrel) [reply rest-server/conf/logback.xml:31]: aha, ok

**Pull request merged**